wrangler-transform/src/main/java/io/cdap/wrangler/WranglerErrorUtil.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright © 2025 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.wrangler;
+
+import com.google.common.base.Throwables;
+import io.cdap.cdap.api.exception.ErrorCategory;
+import io.cdap.cdap.api.exception.ErrorType;
+import io.cdap.cdap.api.exception.ErrorUtils;
+import io.cdap.cdap.api.exception.ProgramFailureException;
+import io.cdap.wrangler.api.DirectiveExecutionException;
+import io.cdap.wrangler.api.DirectiveLoadException;
+import io.cdap.wrangler.api.DirectiveNotFoundException;
+import io.cdap.wrangler.api.DirectiveParseException;
+import io.cdap.wrangler.api.RecipeException;
+import io.cdap.wrangler.expression.ELException;
+import io.cdap.wrangler.utils.RecordConvertorException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Error util file to handle exceptions caught in Wrangler plugin
+ */
+public final class WranglerErrorUtil {
+
+  private static final Map<Class<? extends Throwable>, String> TERMINAL_EXCEPTIONS;
+  private static final Map<Class<? extends Throwable>, String> NON_TERMINAL_EXCEPTIONS;
+
+  static {
+    TERMINAL_EXCEPTIONS = new HashMap<>();
+    TERMINAL_EXCEPTIONS.put(DirectiveParseException.class, "Parsing-Directive");
+    TERMINAL_EXCEPTIONS.put(DirectiveExecutionException.class, "Executing-Directive");
+    TERMINAL_EXCEPTIONS.put(DirectiveLoadException.class, "Loading-Directive");
+    TERMINAL_EXCEPTIONS.put(DirectiveNotFoundException.class, "NotFound-Directive");
+    TERMINAL_EXCEPTIONS.put(RecordConvertorException.class, "RecordConvertor");
+    TERMINAL_EXCEPTIONS.put(ELException.class, "ExpressionLanguage-Parsing");
+    TERMINAL_EXCEPTIONS.put(PreconditionException.class, "Precondition");
+
+    NON_TERMINAL_EXCEPTIONS = new HashMap<>();
+    NON_TERMINAL_EXCEPTIONS.put(RecipeException.class, "Executing-Recipe");
+  }
```

Comment:
In Java, we generally create an immutable map instead using a hashMap, if it is not meant to be modified later:

```
private static final Map<String, String> TERMINAL_EXCEPTIONS = ImmutableMap.<String, String>builder()
            .put(DirectiveParseException.class.getName(), "Parsing-Directive")
            ........
            .build();

private static final Map<String, String> NON_TERMINAL_EXCEPTIONS = ImmutableMap.<String>, String>builder()
            .put(RecipeException.class.getName(), "Executing-Recipe")
            .build();
```